api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -645,8 +645,10 @@ public void changePassword(User user, String oldPassword, String newPassword) th
 			}
 		} else if (!dao.getLoginCredential(user).checkPassword(oldPassword)) {
 			throw new APIException("old.password.not.correct", (Object[]) null);
+
+		} else if (dao.getLoginCredential(user).checkPassword(newPassword)) {
+			throw new APIException("new.password.equal.to.old", (Object[]) null);
```

Comment:
This is perfectly fine. However, at this point, we've verified:

1. That user exists
2. That the old password is given and correct

So it might be simpler to replace this with:

```java
else if (oldPassword.equals(newPassword) {
```

Also, we should probably check that the `newPassword` isn't null.

Reply:
Thanks @ibacher this is far better.